src/Algebra/Structures.agda

```diff
@@ -425,6 +425,15 @@ record IsCommutativeSemiring (+ * : Op₂ A) (0# 1# : A) : Set (a ⊔ ℓ) where
     }
 
 
+record IsCommutativeCancellativeSemiring (+ * : Op₂ A) (0# 1# : A) : Set (a ⊔ ℓ)
+  where
+  field isCommutativeSemiring : IsCommutativeSemiring + * 0# 1#
+
+  open IsCommutativeSemiring isCommutativeSemiring public
```

Comment:
No need to put the `open` statement in the middle of the fields, can go at the end.

Reply:
I am going to implement all your today notes.
And I have a couple of questions.
1) In Algebra.Definitions.AlmostLeftCancellative  it is desirable to use  ≉,  but I do not find where to import it from
     without obtaining cyclic module dependency.

2) In principle, it needs to be first CancellativeSemiring, and then, CancellativeCommutativeSemiring on top of it
    (with adding the *-comm axiom).
I skip the former, because a) I currently deal only with commutative case,  b) the PR goal is GCD, and standard library needs not theoretical complications related to GCD for a non-commutative case, and I do not know anything about
GCD for a non-commutative case.
Do we need to declare  CancellativeSemiring and then CancellativeCommutativeSemiring on top of it? 

3) Imagine CancellativeCommutativeSemiring with decidable equality, decidable divisibility, and trivial mutual divisibility relation. Do we need to call this record
  CancellativeCommutativeSemiring-with≟,DecDiv,trivialMutualDiv ?
 
And it will need three more levels in records, each one with a single property law, and the three Is-structures, and 
Is-structures having repetitions of several fields like, say, carrier.    
?